source/puppet/4.2/reference/modules_installing.markdown

```diff
@@ -51,130 +47,137 @@ If you have used a command line package manager tool (like `gem`, `apt-get`, or
 * Use the `--debug` option to see additional information about what the Puppet module tool is doing.
 
-
->**A note about installing**
+> **A note about installing**
 >
->As of Puppet 3.6, if any module in your /etc/puppetlabs/puppet/modules directory has incorrect versioning (anything other than major.minor.patch), attempting to install a module will result in this warning.
+> As of Puppet 3.6, if any module in your `modules` directory (`/etc/puppetlabs/code/modules` in Puppet 4) has incorrect versioning (anything other than major.minor.patch), attempting to install a module will result in this warning:
 >
->~~~
->Warning: module (/Users/youtheuser/.puppet/modules/module) has an invalid version number (0.1). The version has been set to 0.0.0. If you are the maintainer for this module, please update the metadata.json with a valid Semantic Version (http://semver.org).
-~~~
+> `Warning: module (/Users/youtheuser/.puppet/modules/module) has an invalid version number (0.1). The version has been set to 0.0.0. If you are the maintainer for this module, please update the metadata.json with a valid Semantic Version (http://semver.org).`
 >
->Despite the warning, your module will still be downloaded. The metadata of the offending module will not be altered. The versioning information has only been changed in memory during the run of the program.
-
+> Despite the warning, Puppet still downloads your module and does not permanently change the offending module's metadata. The versioning information has only been changed in memory during the run of the program.
 
 ### Installing From the Puppet Forge
 
 To install a module from the Puppet Forge, simply identify the desired module by its full name. The full name of a Forge module is formatted as username-modulename.
 
-    # puppet module install puppetlabs-apache
+~~~ bash
+sudo puppet module install puppetlabs-apache
+~~~
 
-### Installing From Another Module Repository
+### Installing from Another Module Repository
 
-The module tool can install modules from other repositories that mimic the Forge's interface. To do this, change the [`module_repository`](/references/4.2.latest/configuration.html#modulerepository) setting in [`puppet.conf`](/guides/configuring.html), or specify a repository on the command line with the `--module_repository` option. The value of this setting should be the base URL of the repository; the default value, which uses the Forge, is `https://forgeapi.puppetlabs.com`.
+The module tool can install modules from other repositories that mimic the Forge's interface. To do this, change the [`module_repository`](/references/4.2.latest/configuration.html#modulerepository) setting in [`puppet.conf`](/puppet/4.2.latest/reference/config_file_main.html) or specify a repository on the command line with the `--module_repository` option. The value of this setting should be the base URL of the repository; the default value, which uses the Forge, is `https://forgeapi.puppetlabs.com`.
```

Comment:
This should just be `./config_file_main.html`, so we don't have to change it each time. Also, the 4.2.latest idiom doesn't work in these pages... I still need to unify the handling of that.